libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -1153,6 +1113,89 @@ object MapboxRouteLineUtils {
         }
     }
 
+    internal fun getRestrictedLineExpressionProducer(
+        route: DirectionsRoute,
+        vanishingPointOffset: Double,
+        restrictedSectionColor: Int,
+        activeLegIndex: Int,
+        routeLineColorResources: RouteLineColorResources
+    ): RouteLineExpressionProvider = {
+        val congestionProvider = getAnnotationProvider(route, routeLineColorResources)
+        val trafficExpressionData =
+            getRouteLineTrafficExpressionDataFromCache(route, congestionProvider)
```

Comment:
It might be good to rename `RouteLineTrafficExpressionData`  and all associated methods to something more generic like `RouteLineAnnnotationExpressionData`. It took me some time to understand how restricted roads are related to traffic, and they aren't, we are just computing the data in the same place.

Reply:
Also not a blocker, but let's follow up.